plugins/workspace-backpack/src/backpack.js

```diff
@@ -440,14 +440,20 @@ export class Backpack extends Blockly.DragTarget {
       this.addBlock(/** @type {!Blockly.BlockSvg} */ (dragElement));
     }
   }
+
   /**
-   * Converts the provided block into a cleaned XML string.
+   * Converts the provided block into a JSON string.
    * @param {!Blockly.Block} block The block to convert.
-   * @returns {string} The cleaned XML string.
+   * @returns {string} The JSON object as a string
    * @private
    */
-  blockToCleanXmlString_(block) {
-    return cleanBlockXML(Blockly.Xml.blockToDom(block));
+  blockToJsonString(block) {
+    const json = Blockly.serialization.blocks.save(block);
+    // Add a BLOCK parameter so that the flyout can categorize it
+    json.kind = 'BLOCK';
+    // Does not allow the same block to be added to the backpack
+    delete json['id'];
```

Comment:
Removing the ID looks good! But I think the `blockToJsonString` method still needs two fixes:

1) It should also strip out any of the extraneous info about comments (`width`, `height`, and `pinned`).
2) The IDs & comment data need to be recursively removed from all child blocks as well as the top-level block. So you'll need to do a bit of tree traversing :P